projects/storefrontlib/shared/components/card/card.component.html

```diff
@@ -35,28 +35,39 @@
           {{ content.textBold }}
         </div>
         <div *ngFor="let line of content.text">
-          <div
-            *ngIf="!truncateText; else truncate"
-            class="cx-card-label"
-            [ngClass]="content.customClass"
+          <ng-container
+            *ngTemplateOutlet="
+              showText;
+              context: {
+                content: content,
+                truncateText: truncateText,
+                charactersLimit: charactersLimit,
+                class: 'cx-card-label',
+                text: line
+              }
+            "
           >
-            {{ line }}
-          </div>
-
-          <ng-template #truncate>
-            <cx-truncate-text-popover
-              [charactersLimit]="charactersLimit"
-              [content]="line"
-              [customClass]="content.customClass"
-            ></cx-truncate-text-popover>
-          </ng-template>
+          </ng-container>
         </div>
-        <div class="cx-card-paragraph" *ngFor="let item of content.paragraphs">
+        <div
+          class="cx-card-paragraph"
+          *ngFor="let item of content.paragraphs; trackBy: trackByIndex"
+        >
           <div class="cx-card-paragraph-title">{{ item.title }}</div>
           <div *ngFor="let text of item.text">
-            <div class="cx-card-paragraph-text">
-              {{ text }}
-            </div>
+            <ng-container
+              *ngTemplateOutlet="
+                showText;
+                context: {
+                  content: content,
+                  truncateText: truncateText,
```

Comment:
As we discussed offline, let's introduce a separate indicator for paragraph truncation. There might be a case that `paragrap` needs to be truncated but `label` does not due to its length so we should not make these two places dependent on one flag.

Reply:
Agree! I will take care of it.

Reply:
See [#17969](https://github.com/SAP/spartacus/pull/17969) PR

Reply:
Done!